cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -590,5 +592,58 @@ struct duration_scalar : chrono_scalar<T> {
    */
   typename T::rep count() { return this->value().count(); }
 };
+
+/**
+ * @brief An owning class to represent a list value in device memory
+ */
+class list_scalar : public scalar {
+ public:
+  using value_type = cudf::list_view;
+
+  list_scalar() : scalar(data_type(type_id::LIST)) {}
+  ~list_scalar()                        = default;
+  list_scalar(list_scalar&& other)      = default;
+  list_scalar(list_scalar const& other) = default;
+  list_scalar& operator=(list_scalar const& other) = delete;
+  list_scalar& operator=(list_scalar&& other) = delete;
+
+  /**
+   * @brief Construct a new list scalar object from existing device data
+   *
+   * @param value The value of the list
```

Comment:
I don't think "value" is the right word to use for a collective container like a list.

```suggestion
   * @param elements The elements of the list
```